In [2]:
import pandas as pd
import numpy as np
import pickle

In [3]:
archivo_csv = pd.read_csv("data/exported_db_data.csv")
#archivo_csv.head()
archivo_csv

index         Q10_Part_1       Q10_Part_2 Q10_Part_3 Q10_Part_4  \
0          0                NaN  Colab Notebooks        NaN        NaN   
1          1   Kaggle Notebooks  Colab Notebooks        NaN        NaN   
2          2   Kaggle Notebooks              NaN        NaN        NaN   
3          3                NaN  Colab Notebooks        NaN        NaN   
4          4                NaN              NaN        NaN        NaN   
...      ...                ...              ...        ...        ...   
25968  25968                NaN  Colab Notebooks        NaN        NaN   
25969  25969                NaN              NaN        NaN        NaN   
25970  25970                NaN              NaN        NaN        NaN   
25971  25971                NaN              NaN        NaN        NaN   
25972  25972                NaN              NaN        NaN        NaN   

      Q10_Part_5 Q10_Part_6 Q10_Part_7                           Q10_Part_8  \
0            NaN        NaN        NaN                                  NaN   
1            NaN        NaN        NaN                                  NaN   
2            NaN        NaN        NaN                                  NaN   
3            NaN        NaN        NaN                                  NaN   
4            NaN        NaN        NaN                                  NaN   
...          ...        ...        ...                                  ...   
25968        NaN        NaN        NaN   Amazon Sagemaker Studio Notebooks    
25969        NaN        NaN        NaN                                  NaN   
25970        NaN        NaN        NaN                                  NaN   
25971        NaN        NaN        NaN                                  NaN   
25972        NaN        NaN        NaN                                  NaN   

                   Q10_Part_9 Q10_Part_10           Q10_Part_11  \
0                         NaN         NaN                   NaN   
1                         NaN         NaN                   NaN   
2                         NaN         NaN                   NaN   
3                         NaN         NaN                   NaN   
4                         NaN         NaN  Google Cloud Datalab   
...                       ...         ...                   ...   
25968   Amazon EMR Notebooks          NaN                   NaN   
25969                     NaN         NaN                   NaN   
25970                     NaN         NaN                   NaN   
25971                     NaN         NaN                   NaN   
25972                     NaN         NaN                   NaN   

                                Q10_Part_12 Q10_Part_13 Q10_Part_14  \
0                                       NaN         NaN         NaN   
1                                       NaN         NaN         NaN   
2                                       NaN         NaN         NaN   
3                                       NaN         NaN         NaN   
4                                       NaN         NaN         NaN   
...                                     ...         ...         ...   
25968   Databricks Collaborative Notebooks          NaN         NaN   
25969                                   NaN         NaN         NaN   
25970                                   NaN         NaN         NaN   
25971                                   NaN         NaN         NaN   
25972                                   NaN         NaN         NaN   

      Q10_Part_15 Q10_Part_16 Q10_OTHER  
0             NaN         NaN       NaN  
1             NaN         NaN       NaN  
2             NaN         NaN       NaN  
3             NaN         NaN       NaN  
4             NaN         NaN       NaN  
...           ...         ...       ...  
25968         NaN         NaN       NaN  
25969         NaN        None       NaN  
25970         NaN         NaN       NaN  
25971         NaN        None       NaN  
25972         NaN         NaN       NaN  

[25973

In [4]:
archivo_txt = pd.read_clipboard(sep = ';')
#archivo_txt.head()
archivo_txt.head()
archivo_txt.to_csv("data/archivo_txt.csv")

In [5]:
archivo_csv_txt = pd.read_csv("data/archivo_txt.csv")
archivo_csv_txt

Empty DataFrame
Columns: [Unnamed: 0, /DATOS/attacks.csv]
Index: []

In [6]:
archivo_xml = pd.read_xml("data/data_xml_clean.xml")
#archivo_xml.head(20)
archivo_xml

level_0  index  time    age gender
0            0      0   910  50-54    Man
1            1      1   784  50-54    Man
2            2      2   924  22-24    Man
3            3      3   575  45-49    Man
4            4      4   781  45-49    Man
...        ...    ...   ...    ...    ...
25968    25968  25968  1756  30-34    Man
25969    25969  25969   253  22-24    Man
25970    25970  25970   494  50-54    Man
25971    25971  25971   277  45-49    Man
25972    25972  25972   255  18-21    Man

[25973 rows x 5 columns]

In [7]:
archivo_xml["age"].value_counts()

25-29    4931
18-21    4901
22-24    4694
30-34    3441
35-39    2504
40-44    1890
45-49    1375
50-54     964
55-59     592
60-69     553
70+       128
Name: age, dtype: int64

In [8]:
archivo_kaggle_csv = pd.read_csv("data/data_remaning_kaggle.csv")
#archivo_kaggle_csv.head()
archivo_kaggle_csv

Unnamed: 0  index                                         Q18_Part_1  \
0               0      0  General purpose image/video tools (PIL, cv2, s...   
1               1      1                                                NaN   
2               2      2                                                NaN   
3               3      3  General purpose image/video tools (PIL, cv2, s...   
4               4      4                                                NaN   
...           ...    ...                                                ...   
25968       25968  25968  General purpose image/video tools (PIL, cv2, s...   
25969       25969  25969                                                NaN   
25970       25970  25970                                                NaN   
25971       25971  25971                                                NaN   
25972       25972  25972                                                NaN   

                                              Q18_Part_2  \
0                                                    NaN   
1                                                    NaN   
2                                                    NaN   
3      Image segmentation methods (U-Net, Mask R-CNN,...   
4                                                    NaN   
...                                                  ...   
25968                                                NaN   
25969                                                NaN   
25970                                                NaN   
25971                                                NaN   
25972                                                NaN   

                                              Q18_Part_3  \
0                                                    NaN   
1                                                    NaN   
2                                                    NaN   
3      Object detection methods (YOLOv3, RetinaNet, etc)   
4                                                    NaN   
...                                                  ...   
25968  Object detection methods (YOLOv3, RetinaNet, etc)   
25969                                                NaN   
25970                                                NaN   
25971                                                NaN   
25972                                                NaN   

                                              Q18_Part_4 Q18_Part_5  \
0                                                    NaN        NaN   
1                                                    NaN        NaN   
2                                                    NaN        NaN   
3      Image classification and other general purpose...        NaN   
4                                                    NaN        NaN   
...                                                  ...        ...   
25968  Image classification and other general purpose...        NaN   
25969                                                NaN        NaN   
25970                                                NaN        NaN   
25971                                                NaN        NaN   
25972                                                NaN        NaN   

      Q18_Part_6 Q18_OTHER                                         Q19_Part_1  \
0            NaN       NaN  Word embeddings/vectors (GLoVe, fastText, word...   
1            NaN       NaN                                                NaN   
2            NaN       NaN                                                NaN   
3            NaN       NaN                                                NaN   
4            NaN       NaN                                                NaN   
...          ...       ...                                                ...   
25968        NaN       NaN                                                NaN   
25969        NaN       NaN                                                NaN   
25970        NaN       NaN        

Después de leer los ficheros hemos averiguado que todos tienen el mismo número de filas 25973

5. ¿Cuánto se cobra de media en base al tiempo trabajado/experiencia? (q25)

6. Diferencia de salarios medios entre hombres y mujeres en medio (q25,q2)

10. Tipos de formación, tales como bootcamp, máster, para conseguir un puesto de trabajo relacionado con su puesto de trabajo.

11. Lenguajes de programación prioritarios tanto para hombres como para mujeres.

12. Qué sectores son los que más demandan analítica de datos. (q20,q21,q22)

14. Necesidad de formación para trabajar en senior cuando ya estamos en junior

Hacer preguntas sencillas siempre pero mirar la respuesta en base al genero y en base al pais

In [ ]:
"""Q2--- Porcentaje de genero entrevistado
Q3--- Paises de residencia
Q4--- Nivel de educación de cara a 2 años
Q5--- Rol actual, equivalente
Q6--- Años de experiencia en programación
Q7--- Lenguajes de programación que utilizas
Q8--- Programas que recomiendas aprender a un aspirante de cientifico de datos
Q11-- Plataformas usas como data science
Q14-- Que librerias/herramientas de visualización usas
Q15-- Cuanto tiempo hace que usas métodos de machine learning
Q20-- En qué sector trabajas
Q21-- Tamaño de la compañía(referencia a la q20)
Q22-- Cuanta gente hay en tu departamento(referente a la q20)
Q25-- Salario
"""